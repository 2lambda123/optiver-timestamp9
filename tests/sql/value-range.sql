-- This file consists of tests for checking overflows and validating timestamp9 value range.
set timezone to 'UTC+0';

-- The maximum and minimum values of nanoseconds.
-- 1900-01-01 00:00:00.000000000 UTC+0.
select '-2208988800000000000'::timestamp9;
-- 2261-12-31 23:59:59.999999999 UTC+0.
select '9214646399999999999'::timestamp9;

-- Input out-of-range nanoseconds.
-- 1899-12-31 23:59:59.999999999 UTC+0.
select '-2208988800000000001'::timestamp9;
-- 2262-01-01 00:00:00.000000000 UTC+0.
select '9214646400000000000'::timestamp9;

-- The maximum and minimum values of timestamp9 values.
select '1900-01-01 00:00:00.000000000 UTC+0'::timestamp9;
select '2261-12-31 23:59:59.999999999 UTC+0'::timestamp9;

-- Input out-of-range timestamp9 values.
select '1899-12-31 23:59:59.999999999 UTC+0'::timestamp9;
select '2262-01-01 00:00:00.000000000 UTC+0'::timestamp9;

-- Convert out-of-range TimestampTz values to timestamp9.
select '1899-12-30 23:59:59.999999999 UTC+0'::timestamptz::timestamp9;
select '2262-01-01 00:00:00.000000000 UTC+0'::timestamptz::timestamp9;

-- Convert out-of-range Timestamp values to timestamp9.
select '1899-12-30 23:59:59.999999999'::timestamp::timestamp9;
select '2262-01-01 00:00:00.000000000'::timestamp::timestamp9;

-- Convert out-of-range date values to timestamp9.
select '1899-12-30'::date::timestamp9;
select '2262-01-01'::date::timestamp9;

-- Test various overflows caused by arithmetic calculations on timestamp9 values.
select '1900-01-01 00:00:00.000000000 UTC+0'::timestamp9 - interval '1us';
select '2261-12-31 23:59:59.999999999 UTC+0'::timestamp9 + interval '1us';
select interval '1us' + '2261-12-31 23:59:59.999999999 UTC+0'::timestamp9;
