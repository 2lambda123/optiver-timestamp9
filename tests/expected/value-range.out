-- This file consists of tests for checking overflows and validating timestamp9 value range.
set timezone to 'UTC+0';
-- The maximum and minimum values of nanoseconds.
-- 1700-01-01 00:00:00.000000000 UTC+0.
select '-8520336000000000000'::timestamp9;
             timestamp9              
-------------------------------------
 1700-01-01 00:00:00.000000000 +0000
(1 row)

-- 2261-12-31 23:59:59.999999999 UTC+0.
select '9214646399999999999'::timestamp9;
             timestamp9              
-------------------------------------
 2261-12-31 23:59:59.999999999 +0000
(1 row)

-- Input out-of-range nanoseconds.
-- 1699-12-31 23:59:59.999999999 UTC+0.
select '-8520336000000000001'::timestamp9;
ERROR:  timestamp9 is out of range at character 8
-- 2262-01-01 00:00:00.000000000 UTC+0.
select '9214646400000000000'::timestamp9;
ERROR:  timestamp9 is out of range at character 8
-- The maximum and minimum values of timestamp9 values.
select '1700-01-01 00:00:00.000000000 UTC+0'::timestamp9;
             timestamp9              
-------------------------------------
 1700-01-01 00:00:00.000000000 +0000
(1 row)

select '2261-12-31 23:59:59.999999999 UTC+0'::timestamp9;
             timestamp9              
-------------------------------------
 2261-12-31 23:59:59.999999999 +0000
(1 row)

-- Input out-of-range timestamp9 values.
select '1699-12-31 23:59:59.999999999 UTC+0'::timestamp9;
ERROR:  timestamp9 is out of range at character 8
select '2262-01-01 00:00:00.000000000 UTC+0'::timestamp9;
ERROR:  timestamp9 is out of range at character 8
-- Convert out-of-range TimestampTz values to timestamp9.
select '1699-12-30 23:59:59.999999999 UTC+0'::timestamptz::timestamp9;
ERROR:  overflow happens when converting TimestampTz to timestamp9
select '2262-01-01 00:00:00.000000000 UTC+0'::timestamptz::timestamp9;
ERROR:  overflow happens when converting TimestampTz to timestamp9
-- Convert out-of-range Timestamp values to timestamp9.
select '1699-12-30 23:59:59.999999999'::timestamp::timestamp9;
ERROR:  timestamp9 out of range
select '2262-01-01 00:00:00.000000000'::timestamp::timestamp9;
ERROR:  timestamp9 out of range
-- Convert out-of-range date values to timestamp9.
select '1699-12-30'::date::timestamp9;
ERROR:  date out of range for timestamp9
select '2262-01-01'::date::timestamp9;
ERROR:  date out of range for timestamp9
-- Test various overflows caused by arithmetic calculations on timestamp9 values.
select '1700-01-01 00:00:00.000000000 UTC+0'::timestamp9 - interval '1us';
ERROR:  overflow happens when converting TimestampTz to timestamp9
select '2261-12-31 23:59:59.999999999 UTC+0'::timestamp9 + interval '1us';
ERROR:  overflow happens when converting TimestampTz to timestamp9
select interval '1us' + '2261-12-31 23:59:59.999999999 UTC+0'::timestamp9;
ERROR:  overflow happens when converting TimestampTz to timestamp9
